### PR TITLE
Add missing features to the 'document' page.

### PR DESCRIPTION
**Pull request by OnkarRuikar**
The PR adds members/events that are there in BCD but not in the content.

### CODE REVIEW

File: files/en-us/web/api/document/index.md [hunk 1]
```diff
@@ -113,6 +113,10 @@ _The `Document` interface for HTML documents inherits from the {{DOMxRef("HTMLDo
   - : Gets/sets directionality (rtl/ltr) of the document.
 - {{DOMxRef("Document.domain")}} {{Deprecated_Inline}}
   - : Gets/sets the domain of the current document.
+- {{DOMxRef("Document.fragmentDirective")}} {{ReadOnlyInline}} {{Experimental_Inline}}
+  - : This object may be used to expose additional information about the text fragment or other fragment directives in the future.
```
Comment (teoli2003) [inline files/en-us/web/api/document/index.md:117]: I don't think we should add non-documented features to this list. Adding a lot of red links
 doesn't make the docs any better.
 
Forgotten documented features are ok.

```suggestion
```

File: files/en-us/web/api/document/index.md [hunk 2]
```diff
@@ -125,6 +129,8 @@ _The `Document` interface for HTML documents inherits from the {{DOMxRef("HTMLDo
   - : Sets or gets the title of the current document.
 - {{DOMxRef("Document.URL")}} {{ReadOnlyInline}}
   - : Returns the document location as a string.
+- {{DOMxRef("Document.wasDiscarded")}} {{ReadOnlyInline}} {{Experimental_Inline}}
+  - : Indicates whether the page was discarded while in a hidden tab.
```
Comment (teoli2003) [inline files/en-us/web/api/document/index.md:133]:
```suggestion
```

File: files/en-us/web/api/document/index.md [hunk 3]
```diff
@@ -312,6 +320,8 @@ Listen to these events using `addEventListener()` or by assigning an event liste
   - : Fired when a static {{HTMLElement("script")}} element finishes executing its script
 - {{DOMxRef("Document.beforescriptexecute_event", "beforescriptexecute")}} {{Non-standard_Inline}}
   - : Fired when a static {{HTMLElement("script")}} is about to start executing.
+- {{DOMxRef("SecurityPolicyViolationEvent", "securitypolicyviolation")}}
+  - : Fired when content security policy is violated.
```
Comment (teoli2003) [inline files/en-us/web/api/document/index.md:324]: This is not the correct link:
```suggestion
- {{DOMxRef("Document.securitypolicyviolation_event", "securitypolicyviolation")}}
  - : Fired when a content security policy is violated.
```

File: files/en-us/web/api/document/index.md [hunk 4]
```diff
@@ -372,6 +382,13 @@ Listen to these events using `addEventListener()` or by assigning an event liste
 - {{DOMxRef("Document/keyup_event", "keyup")}}
   - : Fired when a key is released.
 
+### Lifecycle events
+
+- {{DOMxRef("Document/freeze_event", "freeze")}} {{Experimental_Inline}}
+  - : Fired when the document transitions to `FROZEN` state.
+- {{DOMxRef("Document/resume_event", "resume")}} {{Experimental_Inline}}
+  - : Fired when the document transitions from `FROZEN` to `ACTIVE` state.
+
```
Comment (teoli2003) [inline files/en-us/web/api/document/index.md]: Red links. (And the text look weird, I've never seen capitalized keyword like this).
```suggestion
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]
We don't want to add red links. Linking to forgotten pages is ok, but adding red links is not useful. And impossible to review, as it means diving into each feature which is a lot of work for very little usefulness.

File: files/en-us/web/api/document/index.md [hunk 5]
```diff
@@ -372,6 +382,13 @@ Listen to these events using `addEventListener()` or by assigning an event liste
 - {{DOMxRef("Document/keyup_event", "keyup")}}
   - : Fired when a key is released.
 
+### Lifecycle events
+
+- {{DOMxRef("Document/freeze_event", "freeze")}} {{Experimental_Inline}}
+  - : Fired when the document transitions to `FROZEN` state.
+- {{DOMxRef("Document/resume_event", "resume")}} {{Experimental_Inline}}  
+  - : Fired when the document transitions from `FROZEN` to `ACTIVE` state.
+
```
Comment (teoli2003) [inline files/en-us/web/api/document/index.md:391]:
```suggestion
```

Red links…

File: files/en-us/web/api/document/index.md [hunk 6]
```diff
@@ -223,6 +225,8 @@ _This interface also inherits from the {{DOMxRef("Node")}} and {{DOMxRef("EventT
   - : Returns an array of all elements at the specified coordinates.
 - {{DOMxRef("Document.enableStyleSheetsForSet()")}} {{Deprecated_Inline}} {{Non-standard_Inline}}
   - : Enables the style sheets for the specified style sheet set.
+- {{DOMxRef("Document.exitFullscreen()")}}
+  - : Stops document’s fullscreen element from being displayed fullscreen.
```
Comment (OnkarRuikar) [inline files/en-us/web/api/document/index.md:229]:
```suggestion
  - : Stops document's fullscreen element from being displayed fullscreen.
```

**Comment by OnkarRuikar** [comment]
> We don't want to add red links. Linking to forgotten pages is ok, but adding red links is not useful. And impossible to review, as it means diving into each feature which is a lot of work for very little usefulness.

I've reverted those changes.

***

Tested the updated lint workflow in [this commit](https://github.com/mdn/content/pull/20305/commits/c588084f2cb3cd2c9a8e2b317fb2d001d299aa54).

File: files/en-us/web/api/document/index.md (hunk 4 above)
Comment (OnkarRuikar) [reply files/en-us/web/api/document/index.md]: > I've never seen capitalized keyword like this

These are the actual state values in the specification. So I thought mentioning them as it is will be better.

**Review by teoli2003 — APPROVED** [review]
👍

**Pull request merged**